### PR TITLE
PS-4996, PS-3956: Fixing user statistics bugs (8.0)

### PR DESCRIPTION
**Pull request by dutow**
* Added tests to ensure that some untested columns of user/thread statisctics work correctly, as they were mentioned in PS-4996.
  Some counters are only bugged in later versions, but work in 5.6.
* Fixed commit counter: committed transactions weren't counted
* Fixed access denied counter: most access denied paths weren't counted
* Fixed denied connections counter: authentication failures weren't counted

This is a cherry-pick of 6ee45b95be7c931fc83290d83f0c592a065b9deb, as GCA merge in a merge conflict to the 5.7 trunk.

(cherry picked from commit cee57554e8fafc10802773b399af6b392e38c054)

### CODE REVIEW

File: sql/auth/sql_authentication.cc [hunk 1]
```diff
@@ -1174,15 +1174,19 @@ bool auth_plugin_supports_expiration(const char *plugin_name) {
   a helper function to report an access denied error in all the proper places
 */
 static void login_failed_error(THD *thd, MPVIO_EXT *mpvio, int passwd_used) {
+#ifndef EMBEDDED_LIBRARY
+  thd->diff_denied_connections++;
+  update_global_user_stats(thd, false, time(NULL), mpvio->auth_info.user_name,
```
Comment (laurynas-biveinis) [inline sql/auth/sql_authentication.cc]: nullptr

File: sql/auth/sql_authentication.cc [hunk 2]
```diff
@@ -1174,15 +1174,19 @@ bool auth_plugin_supports_expiration(const char *plugin_name) {
   a helper function to report an access denied error in all the proper places
 */
 static void login_failed_error(THD *thd, MPVIO_EXT *mpvio, int passwd_used) {
+#ifndef EMBEDDED_LIBRARY
```
Comment (laurynas-biveinis) [inline sql/auth/sql_authentication.cc]: No EMBEDED_LIBRARY in 8.0

**Review by laurynas-biveinis — CHANGES_REQUESTED** [review]

**Comment by dutow** [comment]
updated.

**Review by laurynas-biveinis — APPROVED** [review]

**Comment by dutow** [comment]
Same fix as for 5.7, jenkins looks good now: https://ps.cd.percona.com/job/percona-server-8.0-param/197/

**Review by laurynas-biveinis — APPROVED** [review]

**Pull request merged**